src/utils/documentProducer/variables.ts

```diff
@@ -81,16 +82,22 @@ export const getRawValue = (variable: VariableWithValues): number | number[] | s
   }
 };
 
-export const missingRequiredFields = (variablesWithValues: VariableWithValues[]) => {
+export const missingRequiredFields = (
+  variablesWithValues: VariableWithValues[],
+  pendingTableValues?: Map<number, VariableTableCell[][]>
+) => {
   const allRequiredVariables = variablesWithValues.filter(
     (v) => v.isRequired && variableDependencyMet(v, variablesWithValues)
   );
-
   const missingRequiredFields = allRequiredVariables.some((variable) => {
     let hasEmptyValue = false;
     const values = variable.values;
     if (!values || values.length === 0 || getRawValue(variable) === undefined || getRawValue(variable) === '') {
       hasEmptyValue = true;
+      if (pendingTableValues?.get(variable.id)?.length) {
```

Comment:
This may be a lot to add, but we may need to do the following:
1) Add isRequired to the `VariableTableCell` type, this can be passed down from `TableVariableWithValues` in `getInitialCellValues`.
2) Check if an entire row is empty. If so, we can treat it like the row does not exist
3) For a row, check if a column is required. If it is, check if the value is not empty
4) Then we need to check a required table has at least one row (which we have here!)

Reply:
If just one cell is needed, we do need to check if at least one cell is non-empty. Basically still step 2 and 4. A user right now can type something, hit backspace and submit an empty string.